vermouth/processors/merge_chains.py

```diff
@@ -47,20 +47,17 @@ def merge_chains(system, chains, all_chains):
     all_chains: bool
         If True, all chains will be merged.
     """
-    if not all_chains and len(chains)>0:
+    if not all_chains and len(chains) > 0:
         _chains = set(chains)
-    elif all_chains and chains is None:
+    elif all_chains and len(chains) == 0:
         _chains = set()
         for molecule in system.molecules:
             # Molecules can contain multiple chains
             _chains.update(node.get('chain') for node in molecule.nodes.values())
     else:
-        LOGGER.warning('Conflicting merging arguments have been given. Will abort merging.')
-        return
+        raise ValueError
```

Comment:
Do we want an informative message here?

Reply:
Have put one, but probably not necessary. Martinize2 will raise its ArgumentError before it gets here if both have been given, so this would have to be someone using the processor on its own for some reason!

Reply:
Exactly that, but that *is* a possibility we want to enable/encourage/support